aten/src/ATen/native/README.md

```diff
@@ -346,6 +346,36 @@ added if applicable), so that it's still available for other backends to use.
 If you implemented a native function in C++ and want to find out which dispatch keyword
 should be used in native_functions.yaml, please [follow steps in dispatch keywords](#choosing-the-right-dispatch-keyword)
 
+### CompositeImplicitAutograd Compliance
+
+Functions registered as CompositeImplicitAutograd MUST work for most, if not
+all, backends. This means that we impose a set of constraints that make it more
+difficult to write a CompositeImplicitAutograd function than writing regular
+PyTorch code.
+
+If you wish to do something that is banned (you may wish to do this for perf
+reasons), please write a backwards formula for your operator so it is no longer
+CompositeImplicitAutograd or hide parts of the operator in a new operator
+that is not CompositeImplicitAutograd.
+
+CompositeImplicitAutograd operators must not:
+- call `set_` or `resize_` or moral equivalents. These are tricky to handle for
```

Comment:
I'm not sure why set_ is here? It is going through the dispatcher and is not more problematic than any other inplace op?
Also it is used by python code quite a bit (nn.Module in particular) so I don't think we want to "encourage" backend writers not to support it.

Reply:
We can remove `set_`. I'm just concerned people will use `set_` as a replacement for `resize_` when writing kernels

Reply:
@albanD I ended up removing `set_` from the list.

Reply:
ok!